products/libllbuild/BuildSystem-C-API.cpp

```diff
@@ -686,10 +687,38 @@ class CAPIExternalCommand : public ExternalCommand {
     return true;
   }
 
+  virtual void startExternalCommand(BuildSystemCommandInterface& bsci,
+                                    core::Task* task) override {
+    cAPIDelegate.start(cAPIDelegate.context,
+                       (llb_buildsystem_command_t*)this,
+                       (llb_buildsystem_command_interface_t*)&bsci,
+                       (llb_task_t*)task);
+  }
+
+  virtual void provideValueExternalCommand(BuildSystemCommandInterface& bsci,
+                                    core::Task* task,
+                                    uintptr_t inputID,
+                                    const BuildValue& value) override {
+
+    // FIXME: Need to figure out how to convert the reference into a BuildValue that CAPIBuildValue can
+    // accept. Use this temporary value for now.
+    BuildValue tmp = BuildValue::makeInvalid();
+    auto value_p = (llb_build_value *)new CAPIBuildValue(std::move(tmp));
```

Comment:
Need to resolve this part first before merging

Reply:
I've dug around and considered.  I think the correct answer here is to allow the CAPI to explicitly create a copy of BuildValue.   It needs to do so, since we are handing the object out to the client, which is free to retain it for an unspecified time frame. 

I've pushed a commit that makes that change.